staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -1035,18 +1029,24 @@ func (r *Request) request(ctx context.Context, fn func(*http.Request, *http.Resp
 //  * http.Client.Do errors are returned directly.
 func (r *Request) Do(ctx context.Context) Result {
 	var result Result
+	// Metrics for total request latency
+	start := time.Now()
 	err := r.request(ctx, func(req *http.Request, resp *http.Response) {
 		result = r.transformResponse(resp, req)
 	})
 	if err != nil {
 		return Result{err: err}
 	}
+	metrics.RequestLatency.Observe(ctx, r.verb, r.finalURLTemplate(), time.Since(start))
```

Comment:
Do you know if `r.finalURLTemplate()` will return the whole URI? If that's the case, we should at least remove the query parameters since they can lead to cardinality issues. The best in my opinion would be to only have the path of the request since it should be good enough for debugging.

Reply:
yeah, I thought the same, a CI run in openshift returns 841 series by URL, seems excessive

https://xzrmvbis-promecieus.apps.cr.j7t7.p1.openshiftapps.com/graph?g0.expr=sum%20by%20(url)%20(rest_client_request_duration_seconds_bucket%7B%7D)&g0.tab=0&g0.stacked=0&g0.show_exemplars=0&g0.range_input=45m51s864ms&g0.end_input=2021-12-09%2016%3A45%3A57&g0.moment_input=2021-12-09%2016%3A45%3A57

However, the request host is important, we need to be able to filter by destination host ... from the client side maybe the path is less important?

Reply:
Indeed the host is important, I wrongly assumed that it would always be the apiserver. 

>  from the client side maybe the path is less important

I'd say that the path is as important since it allows granularity. When defining your SLOs, you might want to skip certain paths / define different SLOs depending on the path that the client is hitting.

So I guess what we want is a combination of the Scheme + Host + Path? Everything except the query params and the fragments really.

Reply:
@aojea following up on our offline discussion, I think we should have one label for the `host` and another for the `path`, so that we won't need to do any post computation to filter based on the path or the host.

Reply:
agree, but then I will wait for your PR to merge https://github.com/kubernetes/kubernetes/pull/106539

Reply:
@aojea I think the kube-apiserver breaks apart the queries by resource and verb.  Does this method break it apart in the same way for comparison?

Reply:
we already have it
https://github.com/kubernetes/kubernetes/blob/15112c7286497cc42b5f0552b8dc5c935c6cac04/staging/src/k8s.io/client-go/rest/request.go#L791-L795

Reply:
I think that you refer to this

ttps://github.com/aojea/kubernetes/blob/1d9d530ee1b672acb9f2ba089123b5350d64dc3b/staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go#L101-L110

 I don't have any inconvenient on trying to standardize and I think is a good idea to be consistent so we can reuse queries, ... , however, the only important field here is the `host:port` of the url, the `path` adds value from the client perspective if we can know that path is being proxied.

We are only interested in the time to get a response to understand if something between the client and the host is acting up. For any other analysis the apiserver audit.log or apiserver metrics has to be the authoritative answer.